src/encoding/rzcobs.rs

```diff
@@ -0,0 +1,267 @@
+// Standard COBS/rCOBS:
+//   00000000 => end of frame
+//   nnnnnnnn => output n-1 bytes from stream, output 0x00
+//   11111111 => output 254 bytes from stream
+//
+// zCOBS/rzCOBS:
+//   00000000 => end of frame
+//   0xxxxxxx => foreach x from LSB to MSB: if x=0 output 1 byte from stream, if x=1 output 0x00
+//   1nnnnnnn => output n+7 bytes from stream, output 0x00
+//   11111111 => output 134 bytes from stream
+
+pub(crate) struct Encoder {
```

Comment:
isn't there a crate we can depend on for the encoding functionality?

Reply:
There's the `rzcobs` crate, but the API is not quite right. The Encoder needs to take ownership of a Writer which is somewhat annoying for Logger impls. Also, it processes stuff byte-by-byte, while handling entire byte slices allows for more optimizations (not in the current code though). 

I thought integrating it here would be easier and more flexible, since it's not that much code after all. Can try to change it to use `rzcobs` if wanted.